pkg/registries/google/transport.go

```diff
@@ -41,16 +41,25 @@ func (t *googleTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	// a) we only need a write lock every hour to refresh the token.
 	// b) refreshing the token multiple times is idempotent.
 	// c) we do not want to block the entire read path for performance reasons.
-	if !concurrency.WithRLock1(&t.mutex, t.token.Valid) {
-		if err := concurrency.WithLock1(&t.mutex, t.refreshNoLock); err != nil {
-			return nil, err
-		}
+	if err := t.ensureValid(); err != nil {
+		return nil, err
 	}
-	return concurrency.WithRLock2(&t.mutex,
+	return concurrency.WithRLock2(t.mutex,
 		func() (*http.Response, error) { return t.Transport.RoundTrip(req) },
 	)
 }
 
+// ensureValid refreshes the access token if it is invalid.
+// oauth2.Token has a hard-coded expiry delta of 10 seconds.
```

Comment:
That might be true for the original token source created by the utils function, but we can use [this constructor](https://pkg.go.dev/golang.org/x/oauth2#ReuseTokenSourceWithExpiry) to configure the expiry time (10 seconds really isn't helpful to be honest, this would likely lead to an immediate authN failure once scanner receives the creds and attempts to fetch the layers).